### ISSUE TITLE
Detect Fn-lock status for vivobook on startupp

### ISSUE THREAD

**Issue by seerge** [seerge/g-helper#1596]
For a while now, after the laptop wakes up from sleep, I have to turn on and off Fn Lock so that secondary functions of F keys work (volume, etc.). It seems that after waking up from sleep/hibernation, the F keys work as normal F keys. I thought it was just some minor quirk of the app so I didn't report or open an issue. However, this issue lead me to think more about why this is happening.

Just for context (from Computer Hope) :

> The FnLk, F-Lock key is a toggle on a PC compatible keyboard used to enable or disable the F1 through F12 keys' secondary functions (hotkeys)...
> Pressing and releasing the F-Lock key toggles on, or enables, the F1-F12 keys to use their standard functions. Pressing the F-Lock key again toggles off (disables) the F1-F12 key standard functions, and turns on the ability to use the secondary functions on those keys.

On my device (Vivobook X515, with dedicated Fn-Lock), if I turn on Fn-Lock via the keyboard (Fn+Esc), G-Helper reports that Fn-Lock is off and vice versa. So it seems, that the Fn-Lock toggle in G-Helper (v0.129) is reversed?

Also, I noticed that upon waking up from sleep, the Fn-Lock on G-Helper stays on. On laptops with a dedicated Fn-Lock with an LED indicator, it is of course turned off. So, I think G-Helper should turn off Fn-Lock upon waking up from sleep/hibernation.

So the flow might be like:

`G-Helper starts up/wakes from sleep/Fn-Lock OFF: UI Fn-Lock OFF (grey) & run this` [command](https://github.com/seerge/g-helper/issues/422#issuecomment-1570895229) `with Control_status=0 (for laptops with a dedicated Fn-Lock) & process the secondary functions of F-keys {if WMI 78 (dedicated Fn-Lock) -> Fn-Lock ON}`

`Fn-Lock ON:  UI Fn-Lock ON (blue) & run this` [command](https://github.com/seerge/g-helper/issues/422#issuecomment-1570895229) `with Control_status=1 & do not process F keys' secondary functions (that is, F keys work normally) but still listen to certain WMI events like 78 {if WMI 78 -> Fn-Lock OFF}`

Thanks!

_Originally posted by @krystean in https://github.com/seerge/g-helper/issues/1581#issuecomment-1806802147_

**Comment by krystean** [comment]
@seerge Sorry about that and thank you!

The output of the command you gave above is:
```
device_status ReturnValue PSComputerName
------------- ----------- --------------
        65536        True
```

The output of the command:
```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DEVS -Arguments @{Device_ID=0x00100023; Control_status=1}
```
which turns on Fn-Lock, that is `Control_status=1`, is:
```
result ReturnValue PSComputerName
------ ----------- --------------
     1        True
```
The `result` is `0` if `Control_status=0`.

Thanks!

**Comment by seerge** [comment]
@krystean good, but what

```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DSTS -Arguments @{Device_ID=0x00100023}
``` 

return when fn-lock is on ?

**Comment by krystean** [comment]
@seerge

The output of the command
```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DSTS -Arguments @{Device_ID=0x00100023}
```

when Fn-Lock is off:
```
device_status ReturnValue PSComputerName
------------- ----------- --------------
        65536        True
```

when Fn-Lock is on:
```
device_status ReturnValue PSComputerName
------------- ----------- --------------
        65536        True
```

It's 65536 when both on and off.

Thanks!

**Comment by krystean** [comment]
@seerge Instead of reading the status of the Fn-Lock from the system, can we instead enforce the status on G-Helper? :) In `config.json` there's an `fn_lock` key that gets updated whenever `Fn-Lock` is toggled on the UI. Is it posible that whenever the value of `fn_lock` is read (eg at startup), the command
```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DEVS -Arguments @{Device_ID=0x00100023; Control_status=<fn_lock_value>}
```
is run using the value of `fn_lock` and whenever the value is updated (eg, turning it on and off in the UI), the command is run with the value it's changed into?

Thanks!

**Comment by seerge** [comment]
@krystean this build will run fn-lock command on app start based on config setting
[GHelper.zip](https://github.com/seerge/g-helper/files/13353671/GHelper.zip)

**Comment by krystean** [comment]
@seerge Thanks! I tried reading the code and made a few changes myself and I think I figured it out:

[changes.txt](https://github.com/seerge/g-helper/files/13356973/changes.txt)

I corrected the ternary operator in the first change. I think I had a hand in this as I remember you had to flip the command when the Fn-Lock was still a checkbox in Extra. Sorry lol

The second change is copying the line you added in [f8df547](https://github.com/seerge/g-helper/commit/f8df547afbe551de21a178e7fea6494a9fd94a87) for init-ing FnLock so that it runs during logon too because Fn keys' alternate functions revert to Windows default ones after waking up from sleep.

Applying these changes on my end make Fn-Lock work as I expect, however, I don't know if it's the best practice lol

Thanks!
